src/celeritas/grid/NonuniformGridCalculator.hh

```diff
@@ -152,11 +169,13 @@ NonuniformGridCalculator::grid() const
  * Make a calculator with x and y flipped.
  *
  * \pre The y values must be monotonic increasing.
+ *
+ * \note Spline interpolation will not be used on an inverse grid.
  */
 CELER_FUNCTION NonuniformGridCalculator
 NonuniformGridCalculator::make_inverse() const
 {
-    return NonuniformGridCalculator{reals_, y_offset_, x_grid_.offset()};
+    return NonuniformGridCalculator{reals_, y_offset_, x_grid_.offset(), {}};
```

Comment:
Maybe add an assertion for not having derivatives? Do we use spline interpolation for the range calculation? (Is it possible for us to be consistent with the interpolated functional representations of dE/dx, range, and inverse range?)

Reply:
Good question: it looks like all three use spline interpolation by default, so with spline enabled we'll probably need to build the inverse range table.

Reply:
Since the range is the integral of the inverse of a piecewise cubic (spline) function, we're not going to be able to exactly represent it with a spline function, right? Maybe it's good enough to use linear interpolation for range and inverse range.

And I would think that since the "inverse range" is really just transposing the x and y axes, it should be straightforward to have an "inverse spline" interpolation?

Reply:
Linear interpolation might be good enough? I guess I don't have a great feel yet for what will be good enough (or, close enough to Geant4).

Hmm I'm not sure how we could do inverse spline interpolation without recalculating the derivatives?